service/controller/v25/cluster_resource_set.go

```diff
@@ -172,6 +173,18 @@ func NewClusterResourceSet(config ClusterResourceSetConfig) (*controller.Resourc
 		}
 	}
 
+	var detectionService *detection.Detection
+	{
+		c := detection.Config{
+			Logger: config.Logger,
+		}
+
+		detectionService, err = detection.New(c)
```

Comment:
Here we create the new abstracted service ready for further use in other resources.